statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,557 @@
+from math import log, pi
+import numpy as np
+from numpy import hstack, vstack
+from numpy.linalg import inv, svd
+import scipy
+import scipy.stats
+from statsmodels.tools.decorators import cache_readonly
+from statsmodels.tools.tools import chain_dot
+from statsmodels.tsa.tsatools import duplication_matrix, vec, vech, unvec
+
+import statsmodels.tsa.base.tsa_model as tsbase
+
+def mat_sqrt(_2darray):
+    u_, s_, v_ = svd(_2darray, full_matrices=False)
+    s_ = np.sqrt(s_)
+    return chain_dot(u_, np.diag(s_), v_)
+
+def _endog_matrices(endog_tot, diff_lags, deterministic):
```

Comment:
add brief docstring to all functions
eventually these will need full docstrings, but can be expanded as the functions or methods and their signature settles down.